plugins/node/instrumentation-runtime-node/src/metrics/eventLoopDelayCollector.ts

```diff
@@ -0,0 +1,215 @@
+        description: metricNames[NodeJsEventLoopDelay.stddev].description,
+        unit: 's',
+      }
+    );
+    const delayp50 = meter.createObservableGauge(
+      `${this.namePrefix}.${NodeJsEventLoopDelay.p50}`,
+      {
+        description: metricNames[NodeJsEventLoopDelay.p50].description,
+        unit: 's',
+      }
+    );
+    const delayp90 = meter.createObservableGauge(
+      `${this.namePrefix}.${NodeJsEventLoopDelay.p90}`,
+      {
+        description: metricNames[NodeJsEventLoopDelay.p90].description,
+        unit: 's',
+      }
+    );
+    const delayp99 = meter.createObservableGauge(
+      `${this.namePrefix}.${NodeJsEventLoopDelay.p99}`,
+      {
+        description: metricNames[NodeJsEventLoopDelay.p99].description,
+        unit: 's',
+      }
+    );
+
+    meter.addBatchObservableCallback(
+      async observableResult => {
+        if (this._scrapeQueue.length === 0) return;
+
+        const data = this._scrapeQueue.shift();
+        if (data === undefined) return;
+
+        const startHrTime = process.hrtime();
+        const delayResult = await new Promise<number>(res => {
+          setImmediate((start: [number, number]) => {
+            res(this._reportEventloopLag(start));
+          }, startHrTime);
+        });
+
+        observableResult.observe(delay, delayResult, this.versionAttribute);
+        observableResult.observe(delayMin, data.min, this.versionAttribute);
+        observableResult.observe(delayMax, data.max, this.versionAttribute);
+        observableResult.observe(delayMean, data.mean, this.versionAttribute);
+        observableResult.observe(
+          delayStddev,
+          data.stddev,
+          this.versionAttribute
+        );
+        observableResult.observe(delayp50, data.p50, this.versionAttribute);
+        observableResult.observe(delayp90, data.p90, this.versionAttribute);
+        observableResult.observe(delayp99, data.p99, this.versionAttribute);
+
+        this._histogram.reset();
+      },
+      [
+        delay,
+        delayMin,
+        delayMax,
+        delayMean,
+        delayStddev,
+        delayp50,
+        delayp90,
+        delayp99,
+      ]
+    );
+  }
+
+  internalEnable(): void {
+    this._histogram.enable();
+  }
+
+  internalDisable(): void {
+    this._histogram.disable();
+  }
+
+  protected scrape(): EventLoopLagInformation {
+    return {
+      min: this.checkNan(this._histogram.min / 1e9),
+      max: this.checkNan(this._histogram.max / 1e9),
+      mean: this.checkNan(this._histogram.mean / 1e9),
+      stddev: this.checkNan(this._histogram.stddev / 1e9),
+      p50: this.checkNan(this._histogram.percentile(90) / 1e9),
+      p90: this.checkNan(this._histogram.percentile(90) / 1e9),
+      p99: this.checkNan(this._histogram.percentile(99) / 1e9),
+    };
+  }
+
+  private _reportEventloopLag(start: [number, number]): number {
+    const delta = process.hrtime(start);
+    const nanosec = delta[0] * 1e9 + delta[1];
+    return nanosec / 1e9;
```

Comment:
**thought:** Why not keep this `ns`? Isn't `process.hrtime` returning the second part because it can't be represented as seconds (within the first part)?

Reply:
Further to this, does it make sense to use the bigint version instead of the method used? The docs indicate this is legacy in node 22. 

Added in node 10:
https://nodejs.org/docs/latest-v10.x/api/process.html#process_process_hrtime_bigint

Reply:
remove this report becouse it's not used in convention